### ISSUE TITLE
Nextcloud instance users don't appear in autocomplete list of users when sharing a folder

### ISSUE THREAD

**Issue by marc0s** [nextcloud/server#8876]
When trying to share a folder to a newly created user, it fails to find it for autocompleting in the form of sharing to a user.

### Steps to reproduce
1. Create a new user
2. Go to files, and try to share a folder by start typing the new user's username in the form

### Expected behaviour
The newly created user's username should appear in the list to select

### Actual behaviour
A message of no users found for 'usern'.

### Server configuration

**Operating system**:
Debian 8

**Web server:**
nginx

**Database:**
postgresql-9.6

**PHP version:**
PHP 5.6.33-0+deb8u1

**Nextcloud version:** (see Nextcloud admin page)
12.0.5

**Updated from an older Nextcloud/ownCloud or fresh install:**
Updated kinda regularly since early owncloud times

**Where did you install Nextcloud from:**
Originally installed from git repo, since the ugprader was available, I used it.

**Signing status:**
<details>
<summary>Signing status</summary>
```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - admin_audit: 1.2.0
  - bruteforcesettings: 1.0.3
  - calendar: 1.5.8
  - comments: 1.2.0
  - contacts: 2.1.2
  - dav: 1.3.1
  - federatedfilesharing: 1.2.0
  - federation: 1.2.0
  - files: 1.7.2
  - files_pdfviewer: 1.1.1
  - files_sharing: 1.4.0
  - files_trashbin: 1.2.0
  - files_videoplayer: 1.1.0
  - gallery: 17.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - nextcloud_announcements: 1.1
  - notifications: 2.0.0
  - oauth2: 1.0.5
  - password_policy: 1.2.2
  - provisioning_api: 1.2.0
  - serverinfo: 1.2.0
  - sharebymail: 1.2.0
  - survey_client: 1.0.0
  - systemtags: 1.2.0
  - tasks: 0.9.6
  - theming: 1.3.0
  - twofactor_backupcodes: 1.1.1
  - twofactor_u2f: 1.4.0
  - updatenotification: 1.2.0
  - workflowengine: 1.2.0
Disabled:
  - activity
  - encryption
  - files_external
  - files_texteditor
  - files_versions
  - firstrunwizard
  - news
  - user_external
  - user_ldap
```
</details>

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:**
Firefox 58.0.1 (64-bit)
**Operating system:**
Debian 9

#### Web server error log
<details>
<summary>Web server error log</summary>

No non-OK entries logged.
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
{"ocs":{"meta":{"status":"ok","statuscode":100,"message":"OK","totalitems":"","itemsperpage":""},"data":{"exact":{"users":[],"groups":[],"remotes":[],"emails":[],"circles":[]},"users":[],"groups":[],"remotes":[],"emails":[],"lookup":[],"circles":[]}}}
```
</details>

**Comment by marc0s** [comment]
Actually, I tried to log in first with that new user, logout, and try again and it neither appeared.

And it's more severe than thought, only contacts from my Contacts appear in such list, no other users from the nextcloud instance appear, even users who I've have already shared files with.

I have set a debug log level in config.php but no info is logged at all when requesting the autocomplete results.

Any hint on where to check something else?

Thanks!

**Comment by marc0s** [comment]
I've just updated the title so it better reflects the actual issue.

**Comment by marc0s** [comment]
Hi, any other information I can give you about this?  It's a really annoying bug and don't know how can I proceed.

Thanks!

**Comment by marc0s** [comment]
I have just updated to v13 and I'm seeing the same problem, no autocompletion candidates when typing a user name's.

**Comment by marc0s** [comment]
Okay... after playing around with the "Sharing" options from the administration page I found the issue: the Catalan translation is wrong. "Restrict" was translated to "Allow", which has a very different meaning.  The point was that I have checked the "Restrict users to only share with users in their groups" which is obviously why I couldn't share anything with users from my instance, where I'm not using groups at all.

Sorry for the noise and I'll be now proposing a good translation for the term.
